File: Object oriented programming/L6/PO_L6_Z1/src/com/company/Node.java
package com.company;

import java.io.Serializable;

public class Node<T>  implements Serializable {
    private T value;
    private Node<T> next;
    private Node<T> prev;

    public Node() // konstruktor node, elementu listy
    {
        next = null;
        prev = null;
        value = null;
    }

    public Node(T val) // konstruktor node, elementu listy
    {
        next = null;
        prev = null;
        value = val;
    }

    public T getValue() { return value; }
    public Node<T> getNext() { return next; }
    public Node<T> getPrev() { return prev; }

    public void setValue(T val) { value = val; }
    public void setNext(Node<T> val) { next = val; }
    public void setPrev(Node<T> val) { prev = val; }
}

File: Object oriented programming/L4/PO_L4_Zad2/PO_L4_Zad2/Node.cs
namespace PO_L4_Zad2
{
    public class Node<T> 
        // obiekty Node są pojedynczymi składnikami listy
    {
        private T value;
        private Node<T> next;
        private Node<T> prev;

        public Node() // konstruktor node, elementu listy
        {
            next = null;
            prev = null;
            value = default(T);
        }

        public T Value // właściwość Value
        {
            get { return this.value; }
            set { this.value = value; }
        }

        public Node<T> Next    // właściwość next
        {
            get { return this.next; }
            set { this.next = value; }
        }
        
        public Node<T> Prev    // właściwość prev
        {
            get { return this.prev; }
            set { this.prev = value; }
        }


    }
}